### ISSUE TITLE
.

### ISSUE THREAD

**Issue by curya** [Mercury13/unicodia#327]
.

**Comment by Mercury13** [comment]
1. Accepted.
2. Please invent some interface, and I’ll try.
3. Interesting. I use SVGs rather than Noto Emoji, some redrawn or from other libraries.

**Comment by Mercury13** [comment]
As I create some funky SVGs ahead of time, there will be some troubles when I mage that SVG, and Noto didn’t yet.

**Comment by curya** [comment]
1. 😎

2. It could be as simple as ctrl+mousewheel. If my mouse is on the left side (Blocks tab/Library tab/etc) then all the glyphs should scale. Likewise, if my mouse is on the right side, then *everything* on the right should scale as well. Currently, ctrl+mousewheel already works on the right side but *only* for the text underneath the glyphs.

3. All the SVGs are on the Github: https://github.com/googlefonts/noto-emoji
Also, it would be nice to *only* use Noto to keep consistency.

**Comment by Mercury13** [comment]
I use that GitHub. I drew some emoji long before Noto came. Some have technical problems, and I fixed them.
Why not just Noto.
1. Technical problems
  - Qt SVG engine does not support clip paths: dancer, ear, some images of superhero/villain
  - SVG engines want some overlap: almost all bicolour and tricolour flags
3. Contrast: placard, dove, tornado, all body parts, some flags with white outline
4. Strange image: cart wheel → Twitter car wheel, families that aren’t marked as silhouettes → older Noto
5. Strange flag toolchain → historically use Twitter

New Noto hands are really nice, but so much work… IDK when I start.

**Comment by curya** [comment]
What about just rendering Noto Color Emoji with the font file?

**Comment by Mercury13** [comment]
I tested, does not work for me for some reason. And contrast, of course.

**Comment by curya** [comment]
Another suggestion:

The ability to turn off showing installed system fonts because sometimes it chooses a font that's just incorrect (even though I have all Noto fonts installed).

Example:
![image](https://github.com/Mercury13/unicodia/assets/71131343/e4505067-02a8-4562-8e9f-2b20ab5bbc5f)

Also notice the spelling error at the bottom.

Should be "Used in." Not "Used un."

**Comment by Mercury13** [comment]
:) How to turn off those fonts?
MYuppy is not system either.

**Comment by curya** [comment]
I mean the option to not display fonts that I've installed myself (fonts outside the Unicodia font folder).

**Comment by Mercury13** [comment]
Those are exactly the fonts installed in the system. I thought about blacklisting some fonts.

**Comment by Mercury13** [comment]
Typo is fixed.

**Comment by curya** [comment]
I'm saying, why are we displaying two different fonts? One is enough.

**Comment by Mercury13** [comment]
One is the sample from Unicodia, controlled by ME (mostly). The other is the sample from your computer, and the font is chosen by some dumb algorithm. At least to ensure that the character is actually supported by computer.

**Comment by curya** [comment]
Right. For me, as someone who has all the Noto fonts installed, showing a font from my computer is useless. The "dumb algorithm" makes it worse.

So a feature to turn of the computer fonts in Unicodia would be nice.

**Comment by Mercury13** [comment]
For example: SIL’s Lanna font is much better than Noto, but it needs a fixup font because of…
1. They chose some specific style, and that language (Khün?) does not have tone 5.
2. Virtual virama is dotted circle 25CC rather than cross.
That’s why I preferred Noto.

Unicodia actually uses some SIL’s fonts: Arabic, Myanmar.

**Comment by Mercury13** [comment]
So, I’m constantly looking for specialized fonts. Right now Unicodia has…
- Arabic, Myanmar, Toto (SIL)
- Coptic (Society for Classical Studies)
- Glagolitic, Banner chant (Project Ponomar)
- Syriac (Beth Mardutho)
- Makasar, Tangsa (right from Unicode requests)
- Sutton (Noto’s glyphs, tables built by me from scratch)
- Brahmi (Segoe, uses Noto just as backup)
- Tibetan (Andrew West)
- Gothic (Junicode)

What I’d like to replace…
- Lanna, Javanese, Aghwank: there are more interesting styles
- Devanagari, Kannada: strange metrics

**Comment by Mercury13** [comment]
@curya 
Working on Favourites.
![image](https://github.com/Mercury13/unicodia/assets/10164882/9df75870-e382-44cd-ba43-60fd18ebdde4)

**Comment by Mercury13** [comment]
@curya 
Now favourites are getting sort of shape.
![image](https://github.com/Mercury13/unicodia/assets/10164882/24af7c42-d6d2-461a-abce-4dff9ac0f622)

**Comment by Mercury13** [comment]
@curya New tasks (probably tonight): synchronize glyph shapes, add to favourites, remove, save to config

**Comment by Mercury13** [comment]
@curya Almost done.
1. I use Tahoma font, and I use a character from DejaVu. It moves layout a few pixels up-down depending on whether DejaVu is present. Does UnicodiaFunky has better dimensions?
2. Save to config.
3. Invent some stub text when the list is empty.

**Comment by Mercury13** [comment]
@curya
![image](https://github.com/Mercury13/unicodia/assets/10164882/1823774d-1fa3-4249-b11f-6553b39ea6ef)

**Comment by Mercury13** [comment]
@curya Favourites done!

**Comment by Mercury13** [comment]
@curya
Zoom: need an interface justification.

B/w emoji: IDK, more work when preparing new Unicode, plus corner cases.

Enlarge sample and remove system font: maybe.

**Comment by Mercury13** [comment]
B/w emoji: sorry, in peacetime. Kyiv is the safest place in Ukraine, but nevertheless I feel unsafe.
I’ve got more business besides looking for b/w emoji and working with size, copyright etc.

**Comment by Mercury13** [comment]
@curya 
B/w emoji: sorry, in peacetime. Kyiv is the safest place in Ukraine, but nevertheless I feel unsafe.
I’ve got more business besides looking for b/w emoji and working with size, copyright etc.

**Comment by curya** [comment]
@Mercury13 Sorry. I've been away for a while.

What's wrong with using mono Noto Emoji?

If it were up to me I would just put the mono and color emoji side by side on the right. And get rid of the system font section since as I said before it's pretty useless.

**Comment by Mercury13** [comment]
> What's wrong with using mono Noto Emoji?

Poor maintainability: while the war is on, I hurry to pull the newest characters somehow. And I don’t ache to draw b/w emoji together with colour.
And because I can disappear anytime, I started to write how to update Unicodia, after all.

> If it were up to me I would just put the mono and color emoji side by side on the right. And get rid of the system font section since as I said before it's pretty useless.

As for me, it’s useful, but I really need to rethink what to do with these not-really-standard fonts.
The main problem of Noto is poor functionality: recently I made Coptic that’s even more functional than Noto. Someone tested three Myanmar fonts, and his one was the winner, but 2nd was SIL Padauk (<10 issues), and 3rd was Noto (dozens of issues).

**Comment by curya** [comment]
I'm not sure I understand. Why is mono Noto Emoji so hard to maintain? All the glyphs are there and they update very rarely...

**Comment by Mercury13** [comment]
New glyphs appear very late.

**Comment by Mercury13** [comment]
About Noto OTF font…

Low contrast
![image](https://github.com/Mercury13/unicodia/assets/10164882/b122cf24-834a-4bfe-b619-ec2775fbc42f)

Not-really-canonical image
![image](https://github.com/Mercury13/unicodia/assets/10164882/97170ad8-3630-4d4a-bfb8-701aa2f7e2b8)

Added very late due to doubts whether hands would be good and what to do, and Slack STILL uses old outlined
![image](https://github.com/Mercury13/unicodia/assets/10164882/8227b2e8-df59-4a51-be66-be1dc8bf1815)

Strange toolchain until recently
![image](https://github.com/Mercury13/unicodia/assets/10164882/3b39acf1-26e6-4dc4-a6ad-ed69a83d8b8e)

**Comment by Mercury13** [comment]
I’m the first who updated THIS flag
![image](https://github.com/Mercury13/unicodia/assets/10164882/71e2bd8d-5d66-4894-be66-440ebb516c23)

**Comment by curya** [comment]
Mono Noto Emoji has no contrast issues and emojis like the wheel look fine to me:
![image](https://github.com/Mercury13/unicodia/assets/71131343/2c7585f7-ff5e-461c-bde7-7a3d7d4a862b)

And flags are simple too:
![image](https://github.com/Mercury13/unicodia/assets/71131343/46fbb652-f378-463e-aeb3-2f147fd82619)

The issue of Noto being late to add new glyphs is not that big of a deal. It's okay to have tofu until new glyphs get added. We aren't in a hurry.

**Comment by Mercury13** [comment]
@curya 
Favourites: OK
Zoom: maybe
Mono emoji: when I’m finally safe, not now. I’m in a big hurry implementing new Unicode because I know: tomorrow I can disappear.
